src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (first == null) {
+            first = last = newNode;
+        } else {
+            this.last.next = newNode;
+            newNode.previous = last;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index out of range: " + index);
+        }
+        Node<T> newNode = new Node<>(value);
+        if (first == null) {
+            first = last = newNode;
+        } else if (index == 0) {
+            newNode.next = first;
+            first = newNode;
+        } else if (index == size) {
+            Node<T> current = last;
+            current.next = newNode;
+            newNode.previous = current;
+            last = newNode;
+        } else {
+            Node<T> current = findNodeIndex(index - 1);
+            newNode.next = current.next;
+            newNode.previous = current;
+            current.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findNodeIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> newNode = findNodeIndex(index);
+        T element = newNode.value;
+        newNode.value = value;
+        return element;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index == 0) {
+            return firstUnlink(findNodeIndex(index));
+        }
+        return unlink(findNodeIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        int index = findIndexObject(object);
+        if (index == 0) {
+            firstUnlink(findNodeIndex(index));
+            return true;
+        } else if (index > 0 && index < size) {
+            unlink(findNodeIndex(index));
+            return true;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return first == null;
+    }
+
+    private Node<T> findNodeIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index out of range: " + index);
+        }
+        Node<T> newNode;
+        newNode = first;
+        for (int i = 0; i < index; i++) {
+            newNode = newNode.next;
+        }
+
+        return newNode;
+    }
```

Comment:
Not fixed
![image](https://user-images.githubusercontent.com/117547096/206014665-4e17c526-fdf0-441e-b866-a7cdd40ed12f.png)